platformio.ini

```diff
@@ -103,11 +104,16 @@ build_flags =
   -D DECODE_LG=true
   -DWLED_USE_MY_CONFIG
 
+lto_flags =
+  -flto
+
 build_unflags =
+  -fno-lto
 
+# No LTO for ESP8266 as we overflow IRAM
 build_flags_esp8266 = ${common.build_flags}  ${esp8266.build_flags}
-build_flags_esp32   = ${common.build_flags}  ${esp32.build_flags}
-build_flags_esp32_V4= ${common.build_flags}  ${esp32_idf_V4.build_flags}
+build_flags_esp32   = ${common.build_flags}  ${esp32.build_flags}         ${common.lto_flags} 
```

Comment:
These are no longer present in current `platformio.ini` so LTO flags should be moved into appropriate `[esp32xx]` sections.
I would also create environment (or better just flags) for debug builds. Enabling all debug flags (as specified in `[common]` section) will no longer link successfully when using anything but plain binary.

Reply:
[platformio.ini.gz](https://github.com/Aircoookie/WLED/files/15161582/platformio.ini.gz)

This is the path I've been going down. I'll rebase against whatever the current development branch is when I get back to it.

Basically, every platform will have a normal (undecorated) and debug environment, with (LTO|None) or DEBUG flags applied as appropriate for the platform.